lib/spack/spack/test/spec_yaml.py

```diff
@@ -505,3 +507,49 @@ def test_load_json_specfiles(specfile, expected_hash, reader_cls):
     # JSON or YAML file, not a list
     for edge in s2.traverse_edges():
         assert isinstance(edge.virtuals, tuple), edge
+
+
+def test_anchorify_1():
+    """Test that anchorify replaces duplicate values with references to a single instance, and
+    that that results in anchors in the output YAML."""
+    before = {"a": [1, 2, 3], "b": [1, 2, 3]}
+    after = {"a": [1, 2, 3], "b": [1, 2, 3]}
```

Comment:
Can we also check something like:
```
{"a": [1, 2, 3], "b": [1, 2]}
```
to check that list items are anchored correctly?